runtime/runtime/src/verifier.rs

```diff
@@ -325,6 +328,21 @@ pub(crate) fn validate_receipt(
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ValidateReceiptMode {
+    /// Used for validating new receipts that were just created.
+    /// More strict than `OldReceipt` mode, which has to handle older receipts.
+    NewReceipt,
+    /// Used for validating older receipts that were saved in the state/received. Less strict than
+    /// NewReceipt validation. Tolerates some receipts that wouldn't pass new validation. It has to
+    /// be less strict because:
+    /// 1) Older receipts might have been created before new validation rules.
+    /// 2) There is a bug which allows to create receipts that are above the size limit. Runtime has
+    ///    to handle them gracefully until the receipt size limit bug is fixed.
+    ///    See https://github.com/near/nearcore/issues/12606 for details.
+    OldReceipt,
```

Comment:
1) I can see that the delayed and incoming receipts are validated but the buffered receipts are not, as far as I can see. Is it an omission? 
2) Is the validation for old receipts actually necessary or is it just an extra protection? Those receipts are a result of on chain computation so I would assume we trust them here already.